packages/skill-api/src/core/init.ts

```diff
@@ -0,0 +1,50 @@
+import {SkillRecordingsOptions} from '../next'
+import {InternalOptions} from './types'
+import {IncomingRequest} from './index'
+
+interface InitParams {
+  host?: string
+  userOptions: SkillRecordingsOptions
+  action: InternalOptions['action']
+  /** Is the incoming request a POST request? */
+  isPost: boolean
+  cookies: IncomingRequest['cookies']
+}
+
+/**
+ * Initialize options and cookies.
+ * @param userOptions
+ * @param action
+ * @param host
+ * @param reqCookies
+ * @param isPost
+ */
+export async function init({
+  userOptions,
+  action,
+  host,
+  cookies: reqCookies,
+  isPost,
+}: InitParams): Promise<{
+  options: InternalOptions
+  cookies: any[]
+}> {
+  // User provided options are overriden by other options,
+  // except for the options with special handling above
+  const options: InternalOptions = {
+    ...userOptions,
+    action,
+    debug: false,
+    pages: {},
+    theme: {
+      colorScheme: 'auto',
+      logo: '',
+      brandColor: '',
+    },
```

Comment:
the theme is for any of the rendered pages. long term this is interesting for 

* signin
* confirmation
* thank you
* check your email
* team management
* invoices
*  ???

The drawback to these pages is that they will have `/api/skill/PAGE_NAME` but the consistency and ability to control this outweighs that in my opinion

We could also consider rewriting these pages as more top level, but for simplicity sake if we started here it'd be very nice.

Reply:
are there any limitations? can we use tailwind and shared components from other packages?